### PR TITLE
Feat/plz tuning max

### LINKED ISSUE #74: Clean programming style in geo_clean
Programming in concise way in the file geo_clean for plz cleaning.

For questions just ask me.

### PR DESCRIPTION
**Pull request by MaxStroh**
implements geo cleaning with geopandas

**IMPORTANT:**
We will need to call "conda install rtree" to be able to use these methods.

Currently in discussion whether we are allowed to use these methods

### CODE REVIEW

**Comment by MaxStroh** [comment]
I will merged the latest dev updates into this branch tomorrow!

File: nextbike/datapreparation/geo_clean.py [hunk 1]
```diff
@@ -33,104 +22,35 @@ def only_nuremberg(p_df):
...
+    gdf_geo_start_pos = gpd.GeoDataFrame(p_df,
+                                         geometry=gpd.points_from_xy(p_df["Longitude_start"],
+                                                                     p_df["Latitude_start"]))
+
+    # ==========
+    # 3. join trips and postalcode dfs on START geo points (to build plz_start)
+    gdf_sjoined_start = gpd.sjoin(gdf_geo_start_pos, gdf_postalcodes, how="inner", op="within")
+    # clean up unnecessary columns added by sjoin, rename plz to plz_start
+    df_with_start_plz = gdf_sjoined_start.drop(["geometry", "index_right", "note"], axis=1)
+    df_with_start_plz.rename({"plz": "plz_start"}, axis=1, inplace=True)
```
Comment (Speedfan94) [inline nextbike/datapreparation/geo_clean.py]: rename plz to postalcode

File: nextbike/datapreparation/geo_clean.py [hunk 2]
```diff
@@ -33,104 +22,35 @@ def only_nuremberg(p_df):
...
+
+    # ==========
+    # 4. create geometry points of END points (with longitude and latitude)
+    gdf_geo_end_pos = gpd.GeoDataFrame(df_with_start_plz,
```
Comment (Speedfan94) [inline nextbike/datapreparation/geo_clean.py]: plz --> postalcoda

File: nextbike/datapreparation/geo_clean.py [hunk 3]
```diff
@@ -33,104 +22,35 @@ def only_nuremberg(p_df):
...
+                                       geometry=gpd.points_from_xy(df_with_start_plz["Longitude_end"],
+                                                                   df_with_start_plz["Latitude_end"]))
+
+    # ==========
+    # 5. join trips and postalcode dfs on END geo points (to build plz_end)
+    gdf_sjoined_end = gpd.sjoin(gdf_geo_end_pos, gdf_postalcodes, how="inner", op="within")
+    # clean up unnecessary columns added by sjoin, rename plz to plz_end
+    df_with_all_plz = gdf_sjoined_end.drop(["geometry", "index_right", "note"], axis=1)
+    df_with_all_plz.rename({"plz": "plz_end"}, axis=1, inplace=True)
```
Comment (Speedfan94) [inline nextbike/datapreparation/geo_clean.py]: plz --> postalcode

File: nextbike/datapreparation/geo_clean.py [hunk 4]
```diff
@@ -33,104 +22,35 @@ def only_nuremberg(p_df):
     # adding plz to df
     # Add PLZ to trip and drop trips without start or end PLZ
```
Comment (Speedfan94) [inline nextbike/datapreparation/geo_clean.py]: Change PLZ To Postalcode

File: nextbike/datapreparation/geo_clean.py [hunk 5]
```diff
@@ -33,104 +22,35 @@ def only_nuremberg(p_df):
...
```
Comment (Speedfan94) [inline nextbike/datapreparation/geo_clean.py]: Add docstring

**Review by Speedfan94 — CHANGES_REQUESTED** [review]
Please add docstrings and rename plz to postalcode, then i am fine :-)

File: nextbike/datapreparation/geo_clean.py (hunk 1 above)
Comment (MaxStroh) [reply nextbike/datapreparation/geo_clean.py]: Done with merge from dev

File: nextbike/datapreparation/geo_clean.py (hunk 2 above)
Comment (MaxStroh) [reply nextbike/datapreparation/geo_clean.py]: Done with merge from dev

File: nextbike/datapreparation/geo_clean.py (hunk 3 above)
Comment (MaxStroh) [reply nextbike/datapreparation/geo_clean.py]: Done with merge from dev

File: nextbike/datapreparation/geo_clean.py (hunk 4 above)
Comment (MaxStroh) [reply nextbike/datapreparation/geo_clean.py]: Done with merge from dev

File: nextbike/datapreparation/geo_clean.py (hunk 5 above)
Comment (MaxStroh) [reply nextbike/datapreparation/geo_clean.py]: Done with merge from dev

**Comment by MaxStroh** [comment]
Done! Please review :D

**Review by Speedfan94 — APPROVED** [review]
all changes approved :-)

File: nextbike/datapreparation/geo_clean.py [hunk 6]
```diff
@@ -1,139 +1,56 @@
...
+    # ==========
+    # 1. load postalcodes data from geojson file
+    path_postalcodes_geojson = io.get_path(p_filename="postleitzahlen-nuremberg.geojson", p_io_folder="input")
+    gdf_postalcodes = gpd.read_file(path_postalcodes_geojson)
+
+    # ==========
+    # 2. create geometry points of START points (with longitude and latitude)
+    gdf_geo_start_pos = gpd.GeoDataFrame(p_df,
+                                         geometry=gpd.points_from_xy(p_df["Longitude_start"],
+                                                                     p_df["Latitude_start"]))
```
Comment (weisske) [inline nextbike/datapreparation/geo_clean.py:34]: Genius

**Review by weisske — APPROVED** [review]
The world will love it

**Review by mulko93 — APPROVED** [review]
nice algorithm and code style

**Pull request merged**